src/components/20-molecules/datepicker/e2e.js

```diff
@@ -3,38 +3,115 @@ import { fixtureURL } from '../../../utils/e2e-helpers';
     test('should not be submit when clicking the arrow keys', async ({
       page,
     }) => {
+      // Navigate to the page with the datepicker
       await page.goto(fixtureURL('examples-datepicker-pure-html--in-a-form'));
 
+      // Fill in the date input field
       await page.fill(
-        `axa-datepicker[data-test-id="datepicker-forms"] .js-datepicker__input`,
+        `[data-test-id="datepicker-forms"] .js-datepicker__input`,
         '29.2.2020'
       );
 
+      // Click on the datepicker input button
       await page.click('.js-datepicker__input-button');
+
+      // Click on the next month arrow button
       await page.click('.js-datepicker__button-next');
 
+      // Check whether the form data does not contain the selected date
       expect(await page.textContent('#form-data-date')).not.toBe(
         '29.02.2020 (of 1 submittable elements)'
       );
     });
+
+    // Test case 3: Check whether the datepicker can add years of a given range
+    test('should just add years of a given range', async ({ page }) => {
+      // Navigate to the page with the datepicker
+      await page.goto(fixtureURL('components-datepicker--datepicker'));
+
+      // Set the allowed years of the datepicker to ['1999-2000']
+      const datepicker = await page.locator('[data-test-id="datepicker"]');
+      await datepicker.evaluate(node => {
+        node.allowedyears = ['1999-2000'];
+      });
+
+      // Get the year dropdown of the datepicker and check whether it contains the correct years
+      const datepickerYearDropdown = await page.locator(
+        '[data-test-id="datepicker"] .js-datepicker__dropdown-year'
+      );
+      const itemsString = await datepickerYearDropdown.getAttribute('items');
+
+      const itemsArray = JSON.parse(itemsString);
+
+      expect(itemsArray.length).toBe(2);
+      expect(itemsArray[0].value).toBe('2000');
+      expect(itemsArray[1].value).toBe('1999');
+      expect(itemsString).not.toContain('"value":"2020"');
+    });
+
+    // Test case 4: Check whether the datepicker fires the right events
+    test('should fire the right events', async ({ page }) => {
+      // Navigate to the page with the datepicker
+      await page.goto(
+        fixtureURL('examples-datepicker-pure-html--with-onchange-handler')
+      );
+
+      // Check whether the datepicker is present
+      const datepickerForm = await page.locator(
+        `[data-test-id="datepicker-onchange"]`
+      );
+      expect(datepickerForm).toBeTruthy();
+
+      // Click on the datepicker input button
+      await page.locator('.js-datepicker__input-button').click();
+
+      // Click on the 16th day of the current month
+      await page
+        .locator('.m-datepicker__calendar-current-month[data-day="16"]')
+        .click();
+
+      // The date object is separated by a newline character in the event log
+      // Therefore, we need to check for the presence of this character to ensure that we are matching the correct date
```

Comment:
But we are testing for **2** successive newlines. So, your comment is incomplete, only half of the story. Once you explain that, it's good.